examples/ProducerConsumer/ProducerConsumer.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"runtime"
+	"runtime/pprof"
+)
+
+type Consumer struct {
+	msgs *chan int
+}
+
+// NewConsumer creates a Consumer
+func NewConsumer(msgs *chan int) *Consumer {
+	return &Consumer{msgs: msgs}
+}
+
+// consume reads the msgs channel
+func (c *Consumer) consume() {
+	fmt.Println("consume: Started")
```

Comment:
To distinguish the outputs from Overlord, Scheduler, and the actual program, we use prefixes in our print/log statements. It would be nice if this program could use some sort of prefix too.
For eg: Instead of "consume:", it would be nice if it could use "[consume]:" or "---consume:". I prefer if it was brackets.

Other than this minor issue, the program looks good!